lib/service/service.go

```diff
@@ -427,26 +427,27 @@ func (process *TeleportProcess) initAuthService(authority sshca.Authority) error
 
 	// first, create the AuthServer
 	authServer, identity, err := auth.Init(auth.InitConfig{
-		Backend:         b,
-		Authority:       authority,
-		ClusterConfig:   cfg.Auth.ClusterConfig,
-		ClusterName:     cfg.Auth.ClusterName,
-		AuthServiceName: cfg.Hostname,
-		DataDir:         cfg.DataDir,
-		HostUUID:        cfg.HostUUID,
-		NodeName:        cfg.Hostname,
-		Authorities:     cfg.Auth.Authorities,
-		ReverseTunnels:  cfg.ReverseTunnels,
-		Trust:           cfg.Trust,
-		Presence:        cfg.Presence,
-		Provisioner:     cfg.Provisioner,
-		Identity:        cfg.Identity,
-		Access:          cfg.Access,
-		StaticTokens:    cfg.Auth.StaticTokens,
-		Roles:           cfg.Auth.Roles,
-		AuthPreference:  cfg.Auth.Preference,
-		OIDCConnectors:  cfg.OIDCConnectors,
-		AuditLog:        process.auditLog,
+		Backend:              b,
+		Authority:            authority,
+		ClusterConfiguration: cfg.ClusterConfiguration,
```

Comment:
I think this breaks teleport because there is no default value passed, and it's passed as nil for teleport OSS

Reply:
it's handled here https://github.com/gravitational/teleport/blob/master/lib/auth/auth.go#L73